salt/_modules/metalk8s_drain.py

```diff
@@ -0,0 +1,512 @@
+        '''
+        controller_ref = _get_controller_of(pod)
+        if controller_ref is None:
+            return None
+        # Fire error if - and only if - controller is gone/missing...
+        # TODO: Is it done
+        #   1) by api call that fires error when nothing is found,
+        #   2) by checking err/warning message and take action?
+        #   3) when reponse.items is empty?
+        #   Note: taking option 3...
+        response = self.get_controller(pod.metadata.namespace, controller_ref)
+        if self.not_found(response):
+            raise DrainException(
+                "Missing pod controller for '{0}'".format(controller_ref.name)
+            )
+        return controller_ref
+
+    def daemonset_filter(self, pod):
+        '''Check if given pod can be elected for deletion daemonset-wise.
+
+        A daemonset will generally not be deleted, regardless of flags, but we
+        can determine if their presence constitutes an error.
+
+        The exception to this rule is when the pod is orphaned, meaning that
+        the management resource is not found, and we have the `force` flag set.
+
+        return: (deletable: bool, warn: str)
+        raises: DrainException if daemonset is in unexpected state
+        '''
+        controller_ref = _get_controller_of(pod)
+
+        if controller_ref is None or controller_ref.kind != "DaemonSet":
+            return True, ""
+
+        controller = self.get_controller(pod.metadata.namespace, controller_ref)
+
+        if self.not_found(controller):
+            if self._force:
+                # Not found and forcing: remove orphan pods with warning
+                return (
+                    True,
+                    "Default to deletable on pod with no controller found"
+                )
+
+            raise DrainException(
+                "Missing pod controller for '{0}'".format(controller_ref.name)
+            )
+
+        if not self._ignore_daemonset:
+            raise DrainException(self.FATAL_MSG["daemonset"])
+
+        return False, self.WARNING_MSG["daemonset"]
+
+    def get_pods_for_eviction(self):
+        '''Get list of pods that can be deleted/evicted on
+        node identified by the name `node_name`.
+        '''
+        warnings = {}
+        failures = {}
+        pods = []
+
+        try:
+            api_instance = kubernetes.client.CoreV1Api()
+            api_response = api_instance.list_pod_for_all_namespaces(
+                field_selector='spec.nodeName={0}'.format(self._node_name)
+            )
+        except (ApiException, HTTPError) as exc:
+            if isinstance(exc, ApiException) and exc.status == 404:
+                return []
+            log.exception(
+                'Exception when calling '
+                'CoreV1Api->list_pod_for_all_namespaces')
+            raise CommandExecutionError(exc)
+
+        for pod in api_response.items:
+            pod_deletable = True
+            for filt in (
```

Comment:
I don't see the benefit of shortening the variable, `filt` doesn't mean anything, and it's not like we care about the built-in `filter` function here:

```suggestion
            for filter in (
```

Reply:
Most syntax highlighters will note that as a keyword though, which we don't want. There's little point in making it that terse though, will expand.

Reply:
> note that as a keyword

which it isn't...

Reply:
Well not in the strictest sense, no, but some syntax highlighters consider it as one.